app/models/spree/payment.rb

```diff
@@ -121,6 +121,12 @@ def actions
       actions
     end
 
+    def resend_authorization_email!
```

Comment:
Where is this method called from? I can't see it anywhere... is it dead code or more metaprogramming?

Reply:
Yep, more metaprogramming - well, more like the same meta programming as the other one: https://github.com/openfoodfoundation/openfoodnetwork/blob/59c981e14d77c20e6d409df3006aa39e4f6d736d/app/models/spree/payment.rb#L113

CreditCard defines an `actions` method that then implies it responds to `can_#{action}` for each action returned.

Reply:
I see there's calls for things like `#can_capture?`, but I still don't see the context where `#resend_authorization_email!` (specifically) would be called?

Reply:
Ah right sorry. That is indeed more metaprogramming, in the `fire` method of the admin/payments_controller.

Reply:
So the value can be `capture`, `void`, `credit`, or `resend_authorization_email`? Is that right? Can it ever be anything different? I see we have some specs for this controller receiving the first three possible events, but not this new one. Maybe we need a spec there? What do you think?

Also, if those four values are the only valid events, it would be really nice to restrict what we pass to `#public_send` to only those four, otherwise in theory anything could be passed here arbitrarily via that param and it will call the method. Like `delete` could be passed and it would call `@payment.delete!`, right?

Reply:
Done and done: 418803e236297d3ce2b5d178bd0c5fdbf6e79b50 and 8c1fc74cb8a83b59e40354348211e3e399786416